File: Frontend/web/src/types/group.ts
export interface GroupListItemResponse {
    id: number;
    name: string;
    courseId: number;
    courseName: string;
    mentorId: number;
    mentorName: string;
    startDate: string;
    endDate: string;
    maxStudents: number;
    activeStudentsCount: number;
    freeSlots: number;
    status: string; // Обычно 'Active', 'Completed', и т.д.
    createdAt: string;
}

// Добавь это в твой файл типов ../types/group
export interface GroupResponse {
    id: number;
    name: string;
    courseId: number;
    courseName: string;
    mentorId: number;
    mentorName: string;
    startDate: string;
    endDate: string;
    maxStudents: number;
    activeStudentsCount: number;
    freeSlots: number;
    status: string;
    createdAt: string;
    schedule?: string | null; // Для кнопки расписания
}

export interface CreateGroupRequest {
    name: string;
    courseId: number;
    mentorId: number;
    startDate: string;
    endDate: string;
    maxStudents: number;
}

export interface UpdateGroupRequest {
    name?: string;
    mentorId?: number;
    startDate?: string;
    endDate?: string;
    maxStudents?: number;
}

File: Backend/src/CrmStella.Domain/Entities/Group.cs
using CrmStella.Domain.Enums;

namespace CrmStella.Domain.Entities;

public class Group
{
    public int Id { get; set; }
    public string Name { get; set; } = null!;
    public int CourseId { get; set; }
    public int MentorId { get; set; }

    public DateTime StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public int MaxStudents { get; set; } = 15;
    public GroupStatus Status { get; set; } = GroupStatus.Active;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? UpdatedAt { get; set; }

    public Course Course { get; set; } = null!;
    public Mentor Mentor { get; set; } = null!;

    public ICollection<GroupStudent> GroupStudents { get; set; } = new List<GroupStudent>();
    public ICollection<Lesson> Lessons { get; set; } = new List<Lesson>();
    public ICollection<Schedule> Schedules { get; set; } = new List<Schedule>();
    public ICollection<Payment> Payments { get; set; } = new List<Payment>();
    public ICollection<Exam> Exams { get; set; } = new List<Exam>();
    public ICollection<WeekResult> WeekResults { get; set; } = new List<WeekResult>();
    public ICollection<StudentProgress> StudentProgresses { get; set; } = new List<StudentProgress>();
}